controller/src/main/java/org/apache/rocketmq/controller/impl/heartbeat/DefaultBrokerHeartbeatManager.java

```diff
@@ -173,4 +175,15 @@ public boolean isBrokerActive(String clusterName, String brokerName, Long broker
         return false;
     }
 
+    @Override
+    public Map<String, Map<String, Long>> getActiveBrokersNum() {
+        Map<String, Map<String, Long>> map = new HashMap<>();
+        this.brokerLiveTable.keySet().forEach(id -> {
+            map.computeIfAbsent(id.getClusterName(), k -> new HashMap<>());
+            map.get(id.getClusterName()).compute(id.getBrokerName(), (broker, num) ->
+                 num == null ? 0L : num + 1L
+            );
+        });
+        return map;
+    }
```

Comment:
- Whether a broker is 'active' should be checked by `isBrokerActive()`, rather than just checking if it is in`brokerLiveTable`.
-   IMO this method provides a dynamic view of `brokerLiveTable`. As the returned data structure is used only in metrics instead of heat beat of brokers, it seems inappropriate to add methods in the  interface `BrokerHeartbeatManager`. Placing it in metric-related packages may be more appropriate.